### ISSUE TITLE
User with viewer role should not be able to create or edit

### ISSUE THREAD

**Issue by grtjn** [marklogic/demo-cat#113]
User info should be extended with role/priv info, and UI should hide create/edit features if user doesn't have appropriate role/privs..

**Comment by ryan321** [comment]
I added a new custom endpoint at /profile which is called on login instead of doing a straight rest search for the profile doc. This endpoint checks the roles on the user and sets "webroles" on the json returned. This endpoint gets the username from the get current web so the username is not a parameter passed in. So only after successful auth would the endpoint code be called, and there's no way to call the endpoint for a user other than the current user (minimizing the risk of accessing someone else's profile.

The code for the node server was changed to handle calling the new endpoint and setting the webroles on the session. All POST and PUT requests go through a function to test that the user has the "writer" webrole. Note that this doesn't prevent the user from directly accessing the MarkLogic server.

The angular code has the webroles set on login and the User Service has a isAdmin() and isWriter() functions as convenience for testing if the 'admin' or 'writer' webroles are on the user.

The Create and Edit demo buttons are shown based on the the result of the isWriter() test.
